packages/bsp/common/usr/lib/armbian/armbian-firstlogin

```diff
@@ -662,6 +662,52 @@ if [[ -f /root/.not_logged_in_yet && -n $(tty) ]]; then
 	echo -e "\nWaiting for system to finish booting ..."
 	systemctl is-system-running --wait > /dev/null
 
+	# enable networkManager-wait-online.service
+	# When NM is used the NetworkManager-wait-online.service must be enabled so that network-online.target is not reached until
+	# NetworkManager has brought up all the interfaces. Service units that require the network to be up before starting rely
+	# on network-online.target working correctly otherwise they will likely fail on boot
+	# Same goes for systemd-networkd stack
+	# https://github.com/armbian/build/issues/7896
+	if
+		systemctl is-enabled --quiet NetworkManager &&
+		systemctl is-enabled --quiet systemd-networkd
+	then
+		echo "Both NetworkManager and systemd-networkd services are enabled."
+		echo "This is known to cause problems with network startup."
+		# If no display manager was found, assume this is a server
```

Comment:
Server and desktop image have NetworkManager, minimal IOT has systemd-networkd. We can't rely on display manager detection, but directly.

Reply:
Can you suggest a test to check directly? I'm not familiar with all the parameters and variations in Armbian. I just made a test based on a comment I saw about NetworkManager Vs systemd-networkd as I was reading posts about the problem of both enabled at the same time.

Reply:
I would just check if systemd-networkd service is active, then this is in use. Network Manager should not be active at that time, and vice versa.

Reply:
Example how we check in armbian-config:
https://github.com/armbian/configng/blob/main/tools/modules/functions/set_runtime_variables.sh#L33-L38
(srv_active is a wrapper function https://github.com/armbian/configng/blob/main/tools/modules/functions/service.sh#L15)

Reply:
OK. I think this issue is resolved by following commit.